test/services/actionsService.test.js

```diff
@@ -1,64 +1,117 @@
-const wreck = require('@hapi/wreck')
-
 let actionsService
-jest.mock('@hapi/wreck')
 
-let payload
+const wreck = { get: jest.fn() }
 
-function stubWreckCall () {
-  wreck.defaults = () => {
-    return {
-      get: () => {
-        return Promise.resolve({ payload })
-      }
-    }
-  }
-}
+describe('actionService', () => {
+  beforeEach(() => {
+    jest.resetModules()
+    wreck.get.mockRestore()
 
-const mockActionServiceResult = {
-  actions: [
-    {
-      id: 'ID1',
-      description: 'Test',
-      rules: [{
-        id: 1,
-        description: 'Test rule',
-        enabled: true,
-        facts: []
-      }]
-    }
-  ]
-}
+    // Setup wreck mocks before requiring actionsService because
+    // actionsService calls wreck.defaults() immediately on load
+    jest.mock('@hapi/wreck')
+    const wreckConstructor = require('@hapi/wreck')
+    wreckConstructor.defaults.mockReturnValue(wreck)
 
-describe('actionService', () => {
-  beforeAll(() => {
-    // Test follows the structure of parcels-service.test.js from ffc-ce-web, which says:
-    // "I tried to stub the call before each, but only the return setup in the first test
-    // was returned, despite trying a combination of clear/reset mocks so I resorted to
-    // setting the payload to a local variable that can be changed before each test
-    // https://github.com/facebook/jest/issues/7136
-    // issue was fixed in November. but not perhaps not made it into the lastest version,
-    // though there are still requests on the issue above to reopen the bug"
-    stubWreckCall()
     actionsService = require('../../server/services/actionsService')
   })
 
-  test('get actions return JSON', async () => {
-    payload = mockActionServiceResult
-    const result = await actionsService.getActions()
-    expect(result).toBeDefined()
-    expect(result.length).toEqual(1)
-    expect(result[0]).toEqual(mockActionServiceResult.actions[0])
+  afterAll(() => {
+    jest.unmock('@hapi/wreck')
```

Comment:
I don't think there's any need for this; jest will automatically remove mocks after exiting a test file. This actually marks that a module should not be mocked, I think it'd be useful in the context of doing auto mocking.

Reply:
Yes, my mistake. I was copying the way we've generally done tests so far but you are correct. The `unmock` function is intended for disabling a mock for a single test or a subset of tests within a file, not for removing mocks at the end.

Reply:
The `afterAll` has now been removed.